File: app/src/main/java/com/ivanmagda/habito/widget/DetailWidgetRemoteViewsService.java
package com.ivanmagda.habito.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

public class DetailWidgetRemoteViewsService extends RemoteViewsService {

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return (new ListProvider(this, intent));
    }

}

File: app/src/main/java/com/ivanmagda/habito/widget/DetailWidgetRemoteViewsService.kt
package com.ivanmagda.habito.widget

import android.appwidget.AppWidgetManager
import android.content.Intent
import android.widget.RemoteViewsService

class DetailWidgetRemoteViewsService : RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
        intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,
                AppWidgetManager.INVALID_APPWIDGET_ID)
        return ListProvider(this, intent)
    }
}
